### ISSUE TITLE
API to read Windows Event Log

### ISSUE THREAD

**Issue by retomerz** [oshi/oshi#631]
Hello,

It would be great to have a API for reading the Windows Event Log.
In my case I want to read all events for the past seven days from source 'System', 'Security' and 'Application'.

Thanks

**Comment by dbwiddis** [comment]
This isn't really in the scope of OSHI, but you can [read how to do it using JNA here](https://code.dblock.org/2010/09/30/jna-reading-windows-event-log-entries-in-java.html).

**Comment by retomerz** [comment]
@dbwiddis Of course this decision is up to you.
From my point of view the Windows Event Log is (a important) part of the operating system info so I thought OSHI would be the right place for such an API.
Anyway, thanks for the link.

**Comment by dbwiddis** [comment]
I think there could be room for some sort of "system log" API (e.g., `dmesg` on Unix) but the fact is reading such logs are actually pretty trivial using JNA so the overhead in figuring out a cross-platform implementation (and then trying to hold logs in memory, etc.) just seems a stretch.  Do you have a vision for how someone would call this in a cross-platform manner?

**Comment by dbwiddis** [comment]
Looked more at this just now.  The link above shows how to do it in JNA, which as a dependency of OSHI is available to you:
```
EventLogIterator iter = new EventLogIterator("Application");
while(iter.hasNext()) {
    EventLogRecord record = iter.next();
    // do stuff with record
}
```
Using `record.getRecord()` will give you the [`EVENTLOGRECORD`](https://docs.microsoft.com/en-us/windows/desktop/api/winnt/ns-winnt-_eventlogrecord) structure to do whatever you want with.

The other OS's have log files in various locations (e.g., `/var/log` on Linux and FreeBSD, `/private/var/log` on OS X, `/var/adm` on Solaris).  OSHI could read those in with various `FileUtil` commands.

Bottom line:  the API needed to read logs already exists, unless we want to get snazzy with features; and given the presence of many better log viewers out there for various OS's I just don't think this is going to happen.
